src/Nethermind/Nethermind.Network/ProtocolsManager.cs

```diff
@@ -218,6 +219,17 @@ public void AddProtocol(string code, Func<ISession, IProtocolHandler> factory)
 
                     return handler;
                 },
+                [Protocol.NodeData] = (session, version) =>
+                {
+                    var handler = version switch
+                    {
+                        1 => new NodeDataProtocolHandler(session, _serializer, _stats, _syncServer, _logManager),
+                        _ => throw new NotSupportedException($"{Protocol.NodeData}.{version} is not supported.")
```

Comment:
very nitpick We could use a pattern of throwing here (and other protocols - parametrize?) from a sub-method with `[DoesNotReturn]` and `[StackTraceHidden]` attributes to decrease the bytecode size a bit.